helios-client/src/main/java/com/spotify/helios/client/DefaultRequestDispatcher.java

```diff
@@ -83,6 +85,22 @@
     endpointIterator = EndpointIterator.of(endpoints);
     this.executorService = executorService;
     this.user = user;
+
+    if (endpointIterator.hasHttps()) {
+      try {
+        identities = getSshIdentities();
+      } catch (Exception e) {
+        log.debug("Couldn't get identities from ssh-agent", e);
+        log.warn(
```

Comment:
@gimaker @mattnworb This should now happen only once. Open to suggestions on how to improve this message.